feat: [3/n]: Role Based Front End Changes

surbhi-posthog opened the pull request:
## Problem

The invite users option gets hidden at the bottom of the screen. Since this option will likely be used by non-technical users. I'm making changes to show this to users higher up on the screen when the user.

I've also tagged along ordering the SDK list to show HTML snippet on top of the SDK list for non technical users since they are more likely to use HTML than Javascript.


## Changes

| Before | After (for non-techical user) |
|--------|-------|
| ![image](https://github.com/user-attachments/assets/b1321641-df34-4ec2-adfc-54befdbf1450) | ![image](https://github.com/user-attachments/assets/a2d0f946-964e-43fe-b5cf-7199cc71c8c3) |



👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Cloud: Yes

## How did you test this code?

Verified if the changes show up for non-technical users
Verified technical users don't see any changes
Verified changes are applied only to product analytics

surbhi-posthog committed:
user model changes

surbhi-posthog committed:
user model changes

surbhi-posthog committed:
user model changes

surbhi-posthog committed:
user model changes

surbhi-posthog committed:
user model changes

surbhi-posthog committed:
user model changes

surbhi-posthog committed:
user model changes

surbhi-posthog committed:
user model changes

surbhi-posthog committed:
user model changes

surbhi-posthog committed:
user model changes

surbhi-posthog committed:
user api changes and tests

surbhi-posthog committed:
user api changes and tests

surbhi-posthog committed:
user api changes and tests

surbhi-posthog committed:
user api changes and tests

surbhi-posthog committed:
user api changes and tests

surbhi-posthog committed:
user api changes and tests

surbhi-posthog committed:
user api changes and tests

surbhi-posthog committed:
user api changes and tests

surbhi-posthog committed:
invite api changes and tests

surbhi-posthog committed:
invite api changes and tests

surbhi-posthog committed:
delete the old migration file

surbhi-posthog committed:
delete the old migration file

surbhi-posthog committed:
delete the old migration file

surbhi-posthog committed:
delete the old migration file

surbhi-posthog committed:
delete the old migration file

surbhi-posthog committed:
delete the old migration file

surbhi-posthog committed:
delete the old migration file

surbhi-posthog committed:
delete the old migration file

surbhi-posthog committed:
frontend experiement changes

surbhi-posthog committed:
frontend experiement changes

surbhi-posthog committed:
frontend experiement changes

surbhi-posthog committed:
frontend experiement changes

surbhi-posthog committed:
frontend experiement changes

surbhi-posthog committed:
frontend experiement changes

surbhi-posthog committed:
frontend experiement changes

surbhi-posthog committed:
frontend experiement changes

surbhi-posthog committed:
frontend experiement changes

surbhi-posthog committed:
frontend experiement changes

surbhi-posthog committed:
removing changes to urlToAction

surbhi-posthog committed:
removing changes to urlToAction

surbhi-posthog committed:
removing changes to urlToAction

surbhi-posthog committed:
removing changes to urlToAction

surbhi-posthog committed:
removing changes to urlToAction

surbhi-posthog committed:
removing changes to urlToAction

surbhi-posthog committed:
removing changes to urlToAction

surbhi-posthog committed:
removing changes to urlToAction

zlwaterfield commented on frontend/src/scenes/userLogic.ts:
```diff
@@ -250,6 +250,14 @@ export const userLogic = kea<userLogicType>([
+        isUserNonTechnical: [
+            (s) => [s.user],
+            (user) => {
+                const technicalRoles = ['engineering', 'founder']
```
I'd argue `data` and `product` are technical so it's just `sales`, `marketing`, `leadership` and `other` that are not.

zlwaterfield replied:
Given we want to keep the experience similar for most users and roll this out to the most non technical users I'd argue we don't need to follow the ICP score exactly. We can also expand it over time but it may be better to limit the exposure of this feature to start.

zlwaterfield commented on posthog/api/shared.py:
```diff
@@ -29,6 +29,7 @@ class Meta:
             "email",
             "is_email_verified",
             "hedgehog_config",
+            "role_at_organization",
```
Can you rebase so all these backend changes aren't here?

zlwaterfield commented on frontend/src/scenes/onboarding/sdks/SDKs.tsx:
```diff
@@ -66,6 +67,14 @@ export function SDKs({
         }
     }, 2000)
 
+    const HelpCard = (): JSX.Element => (
```
Nit: `HelpCard` isn't very descriptive, something along the lines of `InviteTeamMemberCard` might be better

zlwaterfield commented on frontend/src/scenes/onboarding/sdks/SDKs.tsx:
```diff
@@ -66,6 +67,14 @@ export function SDKs({
 
+    const HelpCard = (): JSX.Element => (
+        <LemonCard className={`${isUserInNonTechnicalTest ? 'mb-2' : 'mt-6'}`} hoverEffect={false}>
+            <h3 className="font-bold">Need help with this step?</h3>
```
Less related but maybe we can change this to "Need help?" so it fits better along the first line of the card.

zlwaterfield reviewed (changes requested):
Looks great, just a few comments.

Could you also confirm the role at org is being saved new users?

surbhi-posthog committed:
adding types

surbhi-posthog committed:
adding types

surbhi-posthog committed:
nits and rebase

surbhi-posthog committed:
nits and rebase

surbhi-posthog committed:
manually removing changes to backend files that changes over rebase

surbhi-posthog committed:
manually removing changes to backend files that changes over rebase

surbhi-posthog committed:
comment regarding the techincal roles chosen

surbhi-posthog committed:
comment regarding the techincal roles chosen

surbhi-posthog commented:
Rebased, fixed nits and added comments on why I chose the technical roles here. We need to revisit this when we dig into ICP definition as a follow up.

joshsny commented on frontend/src/scenes/onboarding/sdks/SDKs.tsx:
```diff
@@ -66,6 +67,14 @@ export function SDKs({
         }
     }, 2000)
 
+    const InviteHelpCard = (): JSX.Element => (
```
Perhaps we can pull out the component definition to avoid unnecessary rerenders, and pass `isUserInNonTechnicalTest` as a prop

surbhi-posthog replied:
I'm using sdksLogic to pass in `isUserInNonTechnicalTest` which seems a bit more suitable for using kea patterns

zlwaterfield replied:
+1 this should be outside of the parent and either use a prop or kea hook.

joshsny commented on frontend/src/scenes/onboarding/sdks/SDKs.tsx:
```diff
@@ -106,11 +116,7 @@ export function SDKs({
-                        <p>Invite a team member to help you get set up.</p>
-                        <InviteMembersButton type="secondary" />
-                    </LemonCard>
+                    {!isUserInNonTechnicalTest && <InviteHelpCard />}
```
Does this conflict with the above condition, will one of them always be true?

surbhi-posthog replied:
i'm attempting to show the InviteHelpCard only to people who are in the test group here. `isUserInNonTechnicalTest` has to be true or false based on which we decide on the positioning of the helpcard

joshsny commented on frontend/src/scenes/onboarding/sdks/sdksLogic.tsx:
```diff
@@ -39,10 +42,27 @@ Products that will often be installed in multiple places, eg. web and mobile
+const getProductAnalyticsOrderedSDKs = (sdks: SDK[]): SDK[] => {
+    return [
+        ...sdks.filter((sdk) => sdk.key === 'html'),
+        ...sdks.filter((sdk) => sdk.key === 'javascript-web'),
```
I think this is `javascript_web`, but either way we have an enum we can use in `frontend/src/types.ts` which can help here

joshsny commented on frontend/src/scenes/onboarding/sdks/sdksLogic.tsx:
```diff
@@ -39,10 +42,27 @@ Products that will often be installed in multiple places, eg. web and mobile
 */
 export const multiInstallProducts = [ProductKey.PRODUCT_ANALYTICS, ProductKey.FEATURE_FLAGS]
 
+const getProductAnalyticsOrderedSDKs = (sdks: SDK[]): SDK[] => {
```
Isn't this the default ordering of SDKs already?

joshsny replied:
That makes total sense!

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

surbhi-posthog committed:
fixes

surbhi-posthog committed:
fixes

surbhi-posthog committed:
Merge branch 'onbarding_frontend' of https://github.com/PostHog/posthog into onbarding_frontend

surbhi-posthog commented:
I've updated the isUserNonTechnical logic to include all non-technical users and addressed comments from earlier

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

surbhi-posthog committed:
moving invite card to onboarding and importing into SDK

surbhi-posthog committed:
moving invite card to onboarding and importing into SDK

surbhi-posthog committed:
Merge branch 'onbarding_frontend' of https://github.com/PostHog/posthog into onbarding_frontend

zlwaterfield commented on frontend/src/scenes/onboarding/sdks/SDKs.tsx:
```diff
@@ -15,6 +15,16 @@ import { OnboardingStep } from '../OnboardingStep'
 import { sdksLogic } from './sdksLogic'
 import { SDKSnippet } from './SDKSnippet'
 
+export function InviteHelpCard({ className = '' }: { className?: string }): JSX.Element {
```
Nit: why pass in class name if it's not used?

zlwaterfield reviewed (approved):
One small nit but not a blocker.

surbhi-posthog committed:
addressed  nit

surbhi-posthog committed:
Merge branch 'onbarding_frontend' of https://github.com/PostHog/posthog into onbarding_frontend

surbhi-posthog committed:
Merge branch 'onbarding_frontend' of https://github.com/PostHog/posthog into onbarding_frontend

surbhi-posthog committed:
Merge branch 'master' into onbarding_frontend

surbhi-posthog committed:
Merge branch 'onbarding_frontend' of https://github.com/PostHog/posthog into onbarding_frontend

surbhi-posthog committed:
removing duplication assignment

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)
